### ISSUE TITLE
Add support for "--" option

### ISSUE THREAD

**Issue by atavakoli** [75lb/command-line-args#73]
I'm requesting support for the [POSIX-recommended][1], `--` argument:

[1]: https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html

> The argument ‘--’ terminates all options; any following arguments are treated as non-option arguments, even if they begin with a hyphen.

Encountering a `--` could just add all the following arguments to the option having `defaultOption: true`, or to the `_unknown` property if no `defaultOption` is set and `partial` is true; otherwise, an exception could be thrown.

**Comment by 75lb** [comment]
Hi, yes this functionality is planned for the [next release](https://github.com/75lb/command-line-args/pull/66). I will post more detail about how it will work here once I get sufficient time. Let me know if you have any further ideas or requests for improvement.

**Comment by atavakoli** [comment]
@75lb thanks. Which issue in that PR covers the `--` option?

**Comment by 75lb** [comment]
The idea was introduced by #53 and is still maturing (a WIP).

**Comment by 75lb** [comment]
do you support the idea of having a *stop parsing at first unknown* feature? Some users would prefer if the *first unknown option/value parsed* acted as the `--` style terminator.

In this example, the author could define options for `-v` and `--os` and enable the *stop parsing at first unknown* feature, meaning `ls`,  `-l`  and `-a` would be written to `_unknown`.

```
$ example -v --os centos ls -l -a
```

The above would be equivalent to this but saves the user needing to add `--`:

```
$ example -v --os centos -- ls -l -a
```

**Comment by atavakoli** [comment]
> do you support the idea of having a stop parsing at first unknown feature?

I think this is a fair compromise, as people interested in implementing POSIX-style `--` handling I can just check if the 1st unknown option is `"--"` and ignore it.

Just to be certain, what you're describing is the existing behavior when using `{partial: true}`, right?

**Comment by 75lb** [comment]
no, "stop parsing at first unknown" is a new feature i'm planning to implement. The existing `{partial: true}` behaviour reports all unknown args. The "stop parsing at first unknown" will stop parsing altogether at the first unknown arg it meets, writing off the remaining args as `_unknown` whether they are known or not. 

I'm not suggesting we compromise btw. We can have both "stop parsing at first unknown" and "stop parsing at `--`" as features.

**Comment by atavakoli** [comment]
Ah ok, that makes sense. From my side, I'm happy with whichever (or both) you decide to implement.

**Comment by 75lb** [comment]
does [this explanation of `--`](https://github.com/75lb/command-line-args/wiki/Terminate-parsing-with-a-double-hypen) make sense? 

You can try it using the prerelease.

```
$ npm install command-line-args@next
```

**Comment by 75lb** [comment]
Implemented in [v5.0.0](https://github.com/75lb/command-line-args/releases/tag/v5.0.0).

**Comment by atavakoli** [comment]
Thanks @75lb, and the explanation makes sense (glad you went into detail on how to use it to remove ambiguity).

Looking forward to using it!
